core-bundle/src/Cron/Cron.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Cron;
+
+use Contao\CoreBundle\Monolog\ContaoContext;
+use Doctrine\DBAL\Connection;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+
+class Cron
+{
+    /**
+     * @var string[]
+     */
+    public const INTERVALS = ['monthly', 'weekly', 'daily', 'hourly', 'minutely'];
+
+    /**
+     * @var Connection
+     */
+    protected $db;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @var bool
+     */
+    protected $debug;
+
+    /**
+     * @var array
+     */
+    protected $crons = [];
+
+    public function __construct(Connection $db, LoggerInterface $logger, bool $debug)
+    {
+        $this->db = $db;
+        $this->logger = $logger;
+        $this->debug = $debug;
+    }
+
+    /**
+     * Add a cron service.
+     * 
+     * @param object $service
+     */
+    public function addCronJob($service, string $method, string $interval, int $priority = 0, bool $cli = false): void
+    {
+        if (!\in_array($interval, self::INTERVALS, true)) {
+            throw new \InvalidArgumentException(sprintf('Invalid interval "%s"', $interval));
+        }
+
+        $this->crons[$interval][$priority][] = [$service, $method, $cli];
+    }
+
+    /**
+	 * Run the registered Contao cron jobs.
```

Comment:
Tabs and spaces are mixed all over the file. Running `vendor/bin/php-cs-fixer fix core-bundle/src/Cron` should fix this I think.

Reply:
Fixed.